ils_middleware/tasks/amazon/alma_instance_s3.py

```diff
@@ -0,0 +1,96 @@
+import logging
+from urllib.parse import urlparse
+import os
+from os import path
+from airflow.models import Variable
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+from pymarc import MARCReader
+from rdflib import Graph
+import lxml.etree as ET
+
+logger = logging.getLogger(__name__)
+
+
+def get_from_alma_s3(**kwargs):
+    s3_hook = S3Hook(aws_conn_id="aws_lambda_connection")
+    task_instance = kwargs.get("task_instance")
+    resources = task_instance.xcom_pull(key="resources", task_ids="sqs-message-parse")
+
+    for instance_uri in resources:
+        instance_path = urlparse(instance_uri).path
+        instance_id = path.split(instance_path)[-1]
+
+        temp_file = s3_hook.download_file(
+            key=f"marc/airflow/{instance_id}/record.mar",
+            bucket_name=Variable.get("marc_s3_bucket"),
+        )
+        task_instance.xcom_push(key=instance_uri, value=temp_file)
+
+
+def send_instance_to_alma_s3(**kwargs):
+    s3_hook = S3Hook(aws_conn_id="aws_lambda_connection")
+    task_instance = kwargs.get("task_instance")
+    resources = task_instance.xcom_pull(key="resources", task_ids="sqs-message-parse")
+
+    for instance_uri in resources:
+        instance_path = urlparse(instance_uri).path
+        instance_id = path.split(instance_path)[-1]
+
+        temp_file = task_instance.xcom_pull(
+            key=instance_uri, task_ids="process_alma.download_marc"
+        )
+        marc_record_from_temp_file(instance_id, temp_file)
+        with open(temp_file, "rb") as marc_file:
+            reader = MARCReader(marc_file)
+            for record in reader:
+                instance_field = record.get_fields("884")
+                instance_uri = instance_field[0].get_subfields("k")[0]
+                logger.info(f"Instance URI: {instance_uri}")
+            h = Graph()
+            h.parse(instance_uri)
+            # declare namespaces
+            h.bind("xmlns", "http://www.w3.org/2000/xmlns/")
+            h.bind("xsd", "http://www.w3.org/2001/XMLSchema#")
+            h.bind("rdf", "http://www.w3.org/1999/02/22-rdf-syntax-ns#")
```

Comment:
Not required; `rdf` is one of the default namespaces already bound to a rdflib Graph.